### DISCUSSION TITLE
Integrate custom Kivy Widget

### DISCUSSION THREAD

**Discussion by syllebra** [missionpinball/missionpinball#96] [Q&A]
Hi!

I wrote a custom Kivy widget, working great in regular kivy context. How can I integrate it as custom code? Is this the same way a custom ode for example?

Thank you!

**Comment by toomanybrians** [comment]
Did you write that in Python or that kv script language thing? If you did it in python, I would think you can look through the MPF-MC source (make sure you are looking at MPF-MC, not MPF) and figure it out.

If you write it the kv script, I would guess it won't be easy as MPF-MC doesn't use that at all.

**Comment by syllebra** [comment]
Hi Brian, thank you for your answer.

I actually did it (with some reverse ingineering).
I did not want to make custom dev inside mpf-mc but keep everything in the "machine" folder, as the guys I am working with just installed regular mpf packages.

The only thing I had to do was a custom "run.py" script that can merge a custom spec yaml to the stock one, put that in temp "full" spec yaml and bypass it using a hook in the mpf package.

I think this a missing feature as it is possible to add specific widgets in config file (mpf-mc:    widgets:        custom_shader_widget: criptlets.custom_shader_widget) but the specification validzation fail when using this widget.

I posted more details on the discord channel if you are interssed.

Have a nice day.
